src/manageiq_client/api.py

```diff
@@ -25,24 +25,34 @@ class ManageIQClient(object):
     def __init__(self, entry_point, auth, logger=None, verify_ssl=True, ca_bundle_path=None):
         """ If ca_bundle_path is specified it replaces the system's trusted root CAs"""
         self._entry_point = entry_point
-        if isinstance(auth, dict):
-            self._auth = (auth["user"], auth["password"])
-        elif isinstance(auth, (tuple, list)):
-            self._auth = tuple(auth[:2])
-        else:
-            raise ValueError("Unknown values provider for auth")
         self._session = requests.Session()
+        self._build_auth(auth)
         if not verify_ssl:
             self._session.verify = False
             warnings.filterwarnings('once', message='.*Unverified HTTPS request.*')
         elif ca_bundle_path:
             self._session.verify = ca_bundle_path
-        self._session.auth = self._auth
-        self._session.headers.update({'Content-Type': 'application/json; charset=utf-8'})
         self.logger = logger or logging.getLogger(__name__)
         self.response = None
         self._load_data()
 
+    def _build_auth(self, auth):
```

Comment:
@syncrou one comment on the current code is that there are two code-paths (`len(auth) == 2` and `== 1`) that are way too similar (would need factoring out to a common function).

Also, let's remember that we probably want to support only a single (and possibly explicit) way of authenticating. The fact that we're not supporting both `dict` and `tuple`/`list` is probably for backward compatibility IIRC.

So all in all (to reduce the amount of repeated code and to focus on just a single auth format for new features) I'd suggest:

```python
    def _build_auth(self, auth):
      if isinstance(auth, dict):
        if "user" in auth:      
          self._auth = (auth["user"], auth["password"])
        if "token" in auth:
          headers['X-Auth-Token'] = auth['token']
          self._session.headers.update(headers)
      elif isinstance(auth, (tuple, list)):  # for backward compatibility
          self._auth = tuple(auth[:2])
      else:
          raise ValueError("Unknown values provider for auth")
```

cc @cben